### ISSUE TITLE
aarch64 llvm 14 regression with vectors: Assertion `mutationIsSane(Rule, Query, Mutation) && "legality mutation invalid for match"' failed

### ISSUE THREAD

**Issue by andrewrk** [llvm/llvm-project#56398]
Does *not* affect LLVM 13.0.1.
Tested with LLVM 14.0.6. Have not yet confirmed whether it affects trunk.

Reduced LLVM IR:

```llvm
define internal fastcc i16 @testOneCtzVector.2() {
Entry:
  %0 = call <64 x i16> @llvm.cttz.v64i16(<64 x i16> zeroinitializer, i1 false)
  %1 = trunc <64 x i16> %0 to <64 x i5>
  store <64 x i5> %1, <64 x i5>* null, align 64
  ret i16 0
}

; Function Attrs: nofree nosync nounwind readnone speculatable willreturn
declare <64 x i16> @llvm.cttz.v64i16(<64 x i16>, i1 immarg) #0

attributes #0 = { nofree nosync nounwind readnone speculatable willreturn }
```

Reproduction:

```console
$ ~/local/llvm14-debug/bin/clang -c reduced.ll -target aarch64-linux-unknown
warning: overriding the module target triple with aarch64-unknown-linux-unknown [-Woverride-module]
clang: /home/andy/Downloads/llvm-project-14/llvm/lib/CodeGen/GlobalISel/LegalizerInfo.cpp:198: llvm::LegalizeActionStep llvm::LegalizeRuleSet::apply(const llvm::LegalityQuery&) const: Assertion `mutationIsSane(Rule, Query, Mutation) && "legality mutation invalid for match"' failed.
```

Downstream issue: https://github.com/ziglang/zig/issues/12013

Apologies for not catching this during the RC phase.

**Comment by DavidSpickett** [comment]
This starts from:
```
$ git bisect good
ebc17a0d68208a967fe8e13e1874874228dda622 is the first bad commit
commit ebc17a0d68208a967fe8e13e1874874228dda622
Author: Matt Arsenault <Matthew.Arsenault@amd.com>
Date:   Tue Jul 27 11:08:06 2021 -0400

    GlobalISel: Scalarize unaligned vector stores

    This has the same problems and limitations as the load path.
```

https://github.com/llvm/llvm-project/commit/ebc17a0d68208a967fe8e13e1874874228dda622

@arsenm

**Comment by aemerson** [comment]
Missing vector legalization rules for G_CTTZ and we end up trying to use a scalar rule instead.

**Comment by aemerson** [comment]
Thanks for the bug report. Should be fixed in 9cc1dd209d20

**Comment by andrewrk** [comment]
Thanks for the fix!

Re-opening since this commit has not yet landed in the release/15.x branch. cc @tstellar

**Comment by nikic** [comment]
@tstellar https://github.com/llvm/llvm-project/runs/7537547137?check_suite_focus=true
```
Cmd('git') failed due to: exit code(128)
  cmdline: git fetch https://github.com/llvm/llvm-project issue56398:issue56398
  stderr: 'fatal: couldn't find remote ref issue56398'
```
There's two problems here:
1. We should not try to mirror the branch if it's already on the llvm-project-release-prs repo
2. We currently assume the repo name is llvm-project. (Maybe this is fine generally, just fails for  llvm-project-release-prs in particular)

**Comment by tstellar** [comment]
> @tstellar https://github.com/llvm/llvm-project/runs/7537547137?check_suite_focus=true
> 
> ```
> Cmd('git') failed due to: exit code(128)
>   cmdline: git fetch https://github.com/llvm/llvm-project issue56398:issue56398
>   stderr: 'fatal: couldn't find remote ref issue56398'
> ```
> 
> There's two problems here:
> 
>     1. We should not try to mirror the branch if it's already on the llvm-project-release-prs repo
> 
OK, I'll make this change.

>     2. We currently assume the repo name is llvm-project. (Maybe this is fine generally, just fails for  llvm-project-release-prs in particular)

I fixed this issue.

**Comment by nikic** [comment]
Thanks! Unfortunately we get a new error now (https://github.com/llvm/llvm-project/runs/7549380969?check_suite_focus=true):

```
Cmd('git') failed due to: exit code(128)
  cmdline: git push --force https://***@github.com/llvm/llvm-project-release-prs issue56398:llvm-issue56398
  stderr: 'remote: Permission to llvm/llvm-project-release-prs.git denied to github-actions[bot].
```

**Comment by tstellar** [comment]
> Thanks! Unfortunately we get a new error now (https://github.com/llvm/llvm-project/runs/7549380969?check_suite_focus=true):
> 
> ```
> Cmd('git') failed due to: exit code(128)
>   cmdline: git push --force https://***@github.com/llvm/llvm-project-release-prs issue56398:llvm-issue56398
>   stderr: 'remote: Permission to llvm/llvm-project-release-prs.git denied to github-actions[bot].
> ```

I fixed this in b1dace63b1e64dc501f964ade5aa62bb7ff6f9cf.
